### ISSUE TITLE
Hope to support Chinese language search!

### ISSUE THREAD

**Issue by sakurayun** [squidfunk/mkdocs-material#3882]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [ ] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

Hope to support Chinese language search!

### Use Cases

I see that in the lunr languages project, Chinese language search has been supported!
But mkdocs is still not supported~
Hope to support Chinese search!

**Comment by squidfunk** [comment]
Thanks for reporting. Closing as duplicate:

- https://github.com/mkdocs/mkdocs/issues/2509#issuecomment-882689383
- #3823
- #2330

**Comment by squidfunk** [comment]
Good news! I've added experimental support for Chinese segmentation to the built-in search plugin as part of [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/). It's not yet released, but you can already try the latest `master`. Note that this is currently only available to [sponsors](https://squidfunk.github.io/mkdocs-material/insiders/#how-to-become-a-sponsor).

The following steps are necessary to make it work:

#### Install `jieba`:

Material for MkDocs will automatically detect if `jieba` is installed – you just need to install it:

```
pip install jieba
```

#### Update [search separator](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#search-separator)

A zero-width whitespace (`\u200b`) is used to segment characters in the search index. This means that this character must now be added to the search `separator` in `mkdocs.yml`:

``` yaml
plugins:
  - search:
      separator: "[\\s\\u200b\\-]+"
```

That's it! When you build your documentation, Chinese characters will be automatically segmented. Note that this is an experimental feature, but I'm very interested in feedback. It would be amazing to improve Chinese language support!

<img width="700" alt="Bildschirmfoto 2022-05-04 um 18 20 47" src="https://user-images.githubusercontent.com/932156/166726034-504404ed-df40-402e-925d-ccd126b1600c.png">

**Comment by squidfunk** [comment]
Chinese language support has been released as part of [8.2.13+insiders-4.14.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.14.0). The [blog article](https://squidfunk.github.io/mkdocs-material/blog/2022/chinese-search-support/) sums up my previous comment and explains how to set it up. I'm not proficient in Chinese, so I'm very interested in feedback!
